core/src/main/java/io/kestra/core/http/client/configurations/AbstractAuthConfiguration.java

```diff
@@ -0,0 +1,18 @@
+package io.kestra.core.http.client.configurations;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import lombok.experimental.SuperBuilder;
+import org.apache.hc.client5.http.auth.CredentialsProvider;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", visible = true, include = JsonTypeInfo.As.EXISTING_PROPERTY)
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = BasicAuthConfiguration.class, name = "BASICAUTH"),
```

Comment:
```suggestion
    @JsonSubTypes.Type(value = BasicAuthConfiguration.class, name = "BASIC"),
```

The standard name is BASIC not BASICAUTH

Reply:
An enum would be great here also, to allow validation